HISTORY.md

```diff
@@ -8,6 +8,7 @@
 * Best-efforts recovery ignores CURRENT file completely. If CURRENT file is missing during recovery, best-efforts recovery still proceeds with MANIFEST file(s).
 * In best-efforts recovery, an error that is not Corruption or IOError::kNotFound or IOError::kPathNotFound will be overwritten silently. Fix this by checking all non-ok cases and return early.
 * When `file_checksum_gen_factory` is set to `GetFileChecksumGenCrc32cFactory()`, BackupEngine will compare the crc32c checksums of table files computed when creating a backup to the expected checksums stored in the DB manifest, and will fail `CreateNewBackup()` on mismatch (corruption). If the `file_checksum_gen_factory` is not set or set to any other customized factory, there is no checksum verification to detect if SST files in a DB are corrupt when read, copied, and independently checksummed by BackupEngine.
+* If BackupEngine is set to store backup table files in `shared_checksum`, the checksum value appeared in the backup table filenames is hexadecimally encoded, instead of being plain `uint32_t` value.
```

Comment:
I didn't check whether now always using hex encoding, but `kChecksumAndFileSize`should ideally keep using decimal, since this is the option to maintain previous behavior. In particular, changing it means increased storage space usage on incremental backup after upgrade, because names have changed. I don't think that's excessive code complexity if we track checksums as "file name safe strings" which had different meaning depending on the naming option used.

Reply:
Good point, thanks!